### PR TITLE
Thesis model in Hydranorth

### PR DESCRIPTION
**Pull request by weiweishi**
Added the thesis metadata to Generic File data model to allow Hydranorth to handle thesis object #524
Includes: 
* thesis metadata model
* thesis presenter and form to allow thesis specific fields on displayed records and edit form 
* hide certain fields on the edit form to avoid unnecessary editing as per specification in Hydranorth Data Dictionary .

### CODE REVIEW

File: lib/rdf_vocabularies/ualterms.rb [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
-  property :fedora3uuid
-  property :fedora3handle
-  property :ingestbatch
-  property :year_created
-  property :remote_resource
+class UALTerms < RDF::Vocabulary("http://terms.library.library.ca/identifiers/")
+  term :trid,
+    label: "Computing Science Technical Report ID"
+    
+  term :trid
```
Comment (pbinkley) [inline lib/rdf_vocabularies/ualterms.rb]: Shouldn't this be removed, since it's redundant?

**Comment by pbinkley** [comment]
I haven't been able to finish the testing on this. It looks good though and should be mergable.

**Comment by pgwillia** [comment]
Spec tests passing for me but not passing the README test.  The migration tasks are failing with 
```
RSolr::Error::Http - 500 Internal Server Error
Error: {'responseHeader'=>{'status'=>500,'QTime'=>3},'error'=>{'msg'=>'Error opening new searcher'
```

Likely this is just my environment so I'll try to reset Solr and try again.

File: lib/rdf_vocabularies/ualterms.rb (hunk 1 above)
Comment (weiweishi) [inline lib/rdf_vocabularies/ualterms.rb]: Yes, the duplicated property :trid should be removed.

**Comment by pgwillia** [comment]
New error from ```rake migration:eraitem```
```
{"resource_type"=>["Image"], "creator"=>["Campbell, Sandy"], "contributor"=>[], "description"=>"Because Yakutsk is on permafrost, utilities must be above ground.  The city is heated by hot water boilers in several sectors.  Hot water is carried through pipes like these to heat the city.", "date_created"=>"2014/11/25", "year_created"=>"2014", "license"=>"Creative Commons Attribution-Non-Commercial-No Derivatives 3.0 Unported", "subject"=>["Photographs", "Permafrost Engineering - Yakutsk Russia", "Utilities - above ground", "Large scale hot water heating"], "spatial"=>["Russia, Yakutsk"], "temporal"=>["2014"], "language"=>"English", "fedora3uuid"=>"uuid:07185c11-20f1-445e-990f-db5cfdc71f47", "fedora3handle"=>"http://hdl.handle.net/10402/era.40158", "trid"=>"", "ser"=>"", "ingestbatch"=>"mw22vx48z"}
FAILED: Item uuid:07185c11-20f1-445e-990f-db5cfdc71f47 migration!
You attempted to set the property `description' to a scalar value. However, this property is declared as being multivalued.
```

**Comment by weiweishi** [comment]
Hmm. I thought I've fixed this problem. Maybe locally but forgot to commit:

In lib/tasks/migration.rake line 401: add square brackets to description
like this:

file_attributes = {"resource_type"=>[type], "creator"=>creators,
"contributor"=>contributors, "description"=>*[description]*,
"date_created"=>date, "year_created"=>year_created, "license"=>license,
"subject"=>subjects, "spatial"=>spatials, "temporal"=>temporals,
"language"=>LANG.fetch(language), "fedora3uuid"=>uuid, "fedora3handle" =>
fedora3handle, "trid" => trid, "ser" => ser, "ingestbatch" =>
@ingest_batch_id}

This should fix the problem.
Weiwei



*Weiwei Shi*

Digital Initiative Applications Librarian
University of Alberta Libraries
2-10L Cameron Library
Edmonton, Alberta, Canada T6G 2J8
Phone:(780)492-7802
Fax: (780)248-1209
Email: weiwei.shi@ualberta.ca

**Pull request merged**